app/services/experiment_control_service.rb

```diff
@@ -4,6 +4,10 @@ class ExperimentControlService
   BATCH_SIZE = 100
 
   class << self
+    def logger
+      @logger ||= Rails.logger.child(class: name)
```

Comment:
What is name here? As an aside, instance variables in class methods always throw me for a loop.